File: EasyCTF/nosource.py
import base64
import math

key = "nosource"

def process(a, b):
    length = max(len(a), len(b))
    out = ""
    i = 0

    while i < length:
      ca = ord(a[i % len(a)])
      cb = ord(b[i % len(b)])
      out += chr(ca ^ cb)
      i += 1
    
    return out


flag = 'Fg4GCRoHCQ4TFh0IBxENAE4qEgwHMBsfDiwJRQImHV8GQAwBDEYvV11BCA=='
flag = base64.b64decode(flag)

result = 'eGF1Zm91amt9eW5ncmNuZSBFYWNyQnh6YEN6KndUfjpoL39ueTRMMjMuew=='
result = base64.b64decode(result)
print(result)

print(base64.b64encode(process(flag, key)))


  

File: EasyCTF/nosource.js
 load('base64.js');
 function process(a, b) {
    var len = Math.max(a.length, b.length);
    var out = [];
    for (var i = 0, ca, cb; i < len; i++) {
      ca = a.charCodeAt(i % a.length);
      cb = b.charCodeAt(i % b.length);
      out.push(ca ^ cb);
    }
    return String.fromCharCode.apply(null, out);
  }

  var inputEl = 'xaufoujk}yngrcne EacrBxz`Cz*wT~:h/ny4L23.{'
  var flag = 'Fg4GCRoHCQ4TFh0IBxENAE4qEgwHMBsfDiwJRQImHV8GQAwBDEYvV11BCA==';
  var key = "nosource"
  print(btoa(process(inputEl, key)))

// console.log(Buffer.from('Hello World!').toString('base64'));
// console.log(Buffer.from(b64Encoded, 'base64').toString());